src/rotation/texture_pile.zig

```diff
@@ -73,20 +73,26 @@ pub fn onBlockBreaking(_: ?main.items.Item, _: Vec3f, _: Vec3f, currentData: *Bl
 	if(currentData.data == 0) {
 		currentData.* = .{.typ = 0, .data = 0};
 	} else {
-		currentData.data = currentData.data - 1;
+		currentData.data = @min(currentData.data, currentData.modeData() - 1) - 1;
 	}
 }
 
+fn isThisItem(block: Block, item: main.items.ItemStack) bool {
```

Comment:
This function name is a bit misleading, my first instinct is to read it as "Is this an item?"
I would suggest `isItemBlock`